app/models/web_hook_events_daily_aggregate.rb

```diff
@@ -0,0 +1,53 @@
+# frozen_string_literal: true
+
+class WebHookEventsDailyAggregate < ActiveRecord::Base
+  belongs_to :web_hook
+
+  default_scope { order("created_at DESC") }
+  after_create :aggregate
+
+  def self.purge_old
+    where("created_at < ?", SiteSetting.retain_web_hook_events_period_days.days.ago).delete_all
+  end
+
+  def self.by_day(start_date, end_date, web_hook_id = nil)
+    result = where("date >= ? AND date <= ?", start_date.to_date, end_date.to_date)
+    result = result.where(web_hook_id: web_hook_id) if web_hook_id
+    result
+  end
+
+  def aggregate
```

Comment:
```suggestion
  def aggregate!
```

Since at the end of this you are calling `save!`